src/cag/pattern_discovery.py

```
@@ -0,0 +1,869 @@
+        for pattern_type, detector in self.detectors.items():
+            if pattern_type in self.active_pattern_types:
+                try:
+                    patterns = detector.detect(data, context)
+                    # Filter by confidence threshold
+                    patterns = [
+                        p for p in patterns
+                        if self._meets_confidence_threshold(p.confidence)
+                    ]
+                    all_patterns.extend(patterns)
+                    self.logger.debug(f"Detector {pattern_type.value} found {len(patterns)} patterns")
+                except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)